src/sys/system_v/shm.rs

```diff
@@ -0,0 +1,476 @@
+///
+/// ```no_run
+/// # use nix::errno::Errno;
+/// # use nix::sys::system_v::shm::*;
+/// # use nix::sys::stat::Mode;
+/// #
+/// struct MyData(i64);
+/// const MY_KEY: i32 = 1337;
+///
+/// let mem_segment = Shm::<MyData>::create_and_connect(
+///     MY_KEY,
+///     Mode::S_IRWXU | Mode::S_IRWXG | Mode::S_IRWXO,
+/// )?;
+/// let mut shared_memory = mem_segment.attach(ShmatFlag::empty())?;
+///
+/// // This is writing to the stored [`MyData`] struct
+/// shared_memory.0 = 0xDEADBEEF;
+/// # Ok::<(), Errno>(())
+/// ```
+///
+pub struct SharedMemory<T> {
+    id: c_int,
+    shm: ManuallyDrop<Box<T>>,
+}
+
+impl<T> Deref for SharedMemory<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.shm
+    }
+}
+impl<T> DerefMut for SharedMemory<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.shm
+    }
+}
+
+impl<T> Drop for SharedMemory<T> {
+    fn drop(&mut self) {
+        Self::shmdt(self).expect("SharedMemory detach from SysV IPC");
+    }
+}
+
+impl<T> SharedMemory<T> {
+    /// Performs control operation specified by `cmd` on the System V shared
+    /// memory segment given by `shmid`.
+    ///
+    /// For more information, see [`shmctl(2)`].
+    ///
+    /// # Example
+    ///
+    /// ## Deleting a shared memory segment
+    ///
+    /// ```no_run
+    /// # use nix::errno::Errno;
+    /// # use nix::sys::system_v::shm::*;
+    /// # use nix::sys::stat::Mode;
+    /// #
+    /// struct MyData(i64);
+    /// const MY_KEY: i32 = 1337;
+    ///
+    /// let mem_segment = Shm::<MyData>::create_and_connect(
+    ///     MY_KEY,
+    ///     Mode::S_IRWXU | Mode::S_IRWXG | Mode::S_IRWXO,
+    /// )?;
+    /// let shared_memory = mem_segment.attach(ShmatFlag::empty())?;
+    ///
+    /// let _ = shared_memory.shmctl(ShmctlFlag::IPC_RMID, None)?;
+    /// # Ok::<(), Errno>(())
+    /// ```
+    ///
+    /// [`shmctl(2)`]: https://man7.org/linux/man-pages/man2/shmctl.2.html
+    pub fn shmctl(
+        &self,
+        shm_cmd: ShmctlFlag,
+        buf: Option<&mut shmid_ds>,
+    ) -> Result<c_int> {
+        let buf_ptr: *mut shmid_ds = match buf {
+            Some(ptr) => ptr,
+            None => ptr::null_mut(),
+        };
+        Errno::result(unsafe { libc::shmctl(self.id, shm_cmd.bits(), buf_ptr) })
+    }
+
+    // -- Private --
+
+    /// Performs the reverse of [`SharedMemory::shmat`], detaching the shared memory segment at
+    /// the given address from the address space of the calling process.
+    ///
+    /// This is called automatically on [`Drop`].
+    ///
+    /// For more information, see [`shmdt(2)`].
+    ///
+    /// [`shmdt(2)`]: https://man7.org/linux/man-pages/man2/shmdt.2.html
+    fn shmdt(&self) -> Result<()> {
+        let shmaddr_ref: *const T = &**self;
+        Errno::result(unsafe { libc::shmdt(shmaddr_ref.cast::<c_void>()) })
+            .map(drop)
+    }
+}
+
+libc_bitflags!(
+    /// Valid flags for the third parameter of the function [`shmget`]
```

Comment:
This makes it render correctly:

```suggestion
    /// Valid flags for the third parameter of the function [`Shm::shmget`].
```

Reply:
Forgot to edit that, nice catch. Fixed.